app/Http/Controllers/API/EventController.php

```diff
@@ -161,9 +161,18 @@ public function addVolunteer(Request $request, $idevents)
             $full_name = null;
         }
 
-        // User is null, this volunteer is either anonymous or no user exists.
+        $eventRole = Role::RESTARTER;
+
         if ($request->has('user') && $request->input('user') !== 'not-registered') {
+            // User is null, this volunteer is either anonymous or no user exists.
             $user = $request->input('user');
+
+            if ($user) {
+                $u = User::find($user);
+
+                // A host of a group who is added to an event becomes a host of the event.
+                $eventRole = $u && $u->role == Role::RESTARTER ? Role::RESTARTER : Role::HOST;
```

Comment:
This would make network coordinators and admins hosts of the event, too, right?

I can think of cases where that's not going to match reality (e.g. me, going to an event).  But, probably generally fine.

Reply:
For the case where a host of one group attends an event by another group, this would automatically make them host of that event?  I don't think that should be the case.  It should only turn hosts of the event's group into hosts of the event, no?

Reply:
> This would make network coordinators and admins hosts of the event, too, right?
> 
> I can think of cases where that's not going to match reality (e.g. me, going to an event). But, probably generally fine.

Would it?  They'd have `Role::NETWORK_COORDINATOR` or `Role::ADMIN`.

Reply:
The second point would also fix the first.  This logic isn't well-tested in any case, so I've added a test for it to check that people are/aren't added as hosts appropriately.